visualization/motion_planning_tasks/src/task_list_model.cpp

```diff
@@ -361,107 +362,76 @@ Qt::DropActions TaskListModel::supportedDropActions() const {
 	return Qt::CopyAction | Qt::MoveAction;
 }
 
-AutoAdjustingTreeView::AutoAdjustingTreeView(QWidget* parent) : QTreeView(parent) {
-// consider viewportSizeHint()
-#if QT_VERSION >= QT_VERSION_CHECK(5, 2, 0)
-	setSizeAdjustPolicy(QAbstractScrollArea::AdjustToContents);
-#endif
-}
-
-void AutoAdjustingTreeView::setStretchSection(int section) {
-	stretch_section_ = section;
-	updateGeometry();
-}
+TaskListView::TaskListView(QWidget* parent) : QTreeView(parent) {}
 
-void AutoAdjustingTreeView::setAutoHideSections(const QList<int>& sections) {
-	auto_hide_cols_ = sections;
-	updateGeometry();
+// dropping onto an item, should expand this item
+void TaskListView::dropEvent(QDropEvent* event) {
+	QModelIndex index = indexAt(event->pos());
+	QTreeView::dropEvent(event);
+	if (event->isAccepted())
+		expand(index);
 }
 
-void AutoAdjustingTreeView::setModel(QAbstractItemModel* model) {
-	size_hints_.clear();
+void TaskListView::setModel(QAbstractItemModel* model) {
 	QTreeView::setModel(model);
-
-	updateGeometry();
+	if (header()->count() >= 4) {
```

Comment:
There are some constants like this that I don't really get. Is this set to work nicely for your screen-size or panel position/width?

Reply:
Ok, I assume it's simply the current number of columns we have in the panels

Reply:
Yes, You don't actually have to review the code in detail.
Some feedback on the GUI would be nice though.

I'll rework part of it these days though as discussed with @rhaschke on Tuesday.